controllers/competition_supervisor/competition_supervisor.py

```diff
@@ -33,6 +33,11 @@ def record_video(supervisor: Supervisor, file_path: Path) -> Iterator[None]:
     print("Saving video to {}".format(file_path))
```

Comment:
It's probably worth wrapping this print in the condition too

Reply:
Yeah, I'd be tempted to move the config check above the `mkdir` here too. At that point it miiight be worth extracting a separate function which does the context management? Only if that's actually clearer though.

Reply:
5b4cdce1db32995bc1be52ba57e430c8020f25de